include/sparrow/record_batch.hpp

```diff
@@ -169,6 +196,25 @@ namespace sparrow
         SPARROW_ASSERT_TRUE(check_consistency());
     }
 
+    template <std::ranges::input_range CR>
+        requires std::same_as<std::ranges::range_value_t<CR>, array>
+    record_batch::record_batch(CR&& columns)
+        : m_name_list(std::ranges::size(columns))
+        , m_array_list(to_vector<array>(std::move(columns)))
+    {
+        std::transform(
+            m_array_list.cbegin(),
+            m_array_list.cend(),
+            m_name_list.begin(),
+            [](const array& ar)
+            {
+                return ar.name();
+            }
+        );
+        init_array_map();
```

Comment:
Can we move these logic to dedicated functions which returns the values to put in the members ?

Reply:
I'm not sure we want to do that, this would require an additional memory allocation instead of filling the members in plae. Besides, this transform is used only here. What is the issue with this implementation?

Reply:
IMO it's cleaner to have only a constructor initialization list.

``` c++

std::vector<name_type> get_names(const std::vector<array>& array_list)
{
    const auto names = array_list | std::views::transform(
                    [](const array& ar)
                    {
                        return ar.name();
                    }
    return std::vector<name_type> {names.begin(), names.end()};
}

  template <std::ranges::input_range CR>
        requires std::same_as<std::ranges::range_value_t<CR>, array>
    record_batch::record_batch(CR&& columns)
        : m_array_list(to_vector<array>(std::move(columns)))
        , m_name_list(get_names(m_array_list))
        , 
    {
        init_array_map(); 
```

Reply:
Oh indeed, I did not get that.